src/fairness/print_hierarchy/flux_shares.cpp

```diff
@@ -0,0 +1,182 @@
+              << "\t-h,--help\t\t\tShow this help message\n"
+              << "\t-P DELIMITER"
+              << "\t\tPrint the database hierarchy in a parsable format\n"
+              << "\t-f DB_PATH"
+              << "\t\t\tSpecify location of the flux-accounting database"
+              << std::endl;
+}
+
+void print_csv_header (const std::string& delimiter="|")
+{
+    std::cout << "Account" << delimiter
+              << "Username" << delimiter
+              << "RawShares" << delimiter
+              << "RawUsage"
+              << std::endl;
+}
+
+void print_csv (std::shared_ptr<weighted_tree_node_t> node,
+                const std::string& indent,
+                const std::string& delimiter="|")
+{
+
+    if (node == nullptr)
+        return;
+
+    // print node data
+    if (node->is_user ()) {
+        std::cout << indent << node->get_parent ()-> get_name () << delimiter
+                  << node->get_name () << delimiter
+                  << node->get_shares () << delimiter
+                  << node->get_usage ()
+                  << std::endl;
+    } else {
+        std::cout << indent << node->get_name () << delimiter << delimiter
+                  << node->get_shares () << delimiter
+                  << node->get_usage ()
+                  << std::endl;
+    }
+
+    // recur on subtree
+    int num_children = node->get_num_children ();
+    if (num_children > 0) {
+        for (int i = 0; i < num_children; i++) {
+            print_csv(node->get_child (i), indent + " ", delimiter);
+        }
+    }
+}
+
+void pretty_print_header ()
+{
+    std::cout << std::setw(20) << std::left << "Account"
+              << std::setw(20) << std::right << "Username"
+              << std::setw(20) << std::right << "RawShares"
+              << std::setw(20) << std::right << "RawUsage"
+              << std::endl;
+}
+
+void pretty_print (std::shared_ptr<weighted_tree_node_t> node,
+                   const std::string& indent)
+{
+    std::string name;
+
+    if (node == nullptr)
+        return;
+
+    // print node data
+    if (node->is_user ()) {
+        name = indent + node->get_parent ()->get_name ();
+        std::cout << std::setw(20) << std::left << name
+                  << std::setw(20) << std::right << node->get_name ()
+                  << std::setw(20) << std::right << node->get_shares ()
+                  << std::setw(20) << std::right << node->get_usage ()
+                  << std::endl;
+    } else {
+        name = indent + node->get_name ();
+        std::cout << std::setw(20) << std::left << name
+                  << std::setw(20) << std::right << ""
+                  << std::setw(20) << std::right << node->get_shares ()
+                  << std::setw(20) << std::right << node->get_usage ()
+                  << std::endl;
+    }
+
+    // recur on subtree
+    int num_children = node->get_num_children ();
+    if (num_children > 0) {
+        for (int i = 0; i < num_children; i++) {
```

Comment:
What is the intended order of printing for the accounts?  Should they be sorted alphabetically or by usage/priority?  If priority, do the vertices in this weighted-tree need to have `calc_and_sort_weighted_children` on them?  It isn't clear to me what the default traversal order is here (maybe the order that the DB returned them it?).

Reply:
The order for printing the accounts is based on the order they are fetched from `get_sub_banks()` in **data_reader_db.cpp**. After the root bank is found, subsequent sub banks are fetched via this SQLite statement:

```c++
s_sub_banks = "SELECT bank_table.bank FROM bank_table WHERE parent_bank=?";
```

And associations with this statement:

```c++
s_assoc = "SELECT association_table.username, association_table.shares, "
          "association_table.bank, association_table.job_usage "
          "FROM association_table WHERE association_table.bank=?"
          "ORDER BY association_table.username";
```

I believe SQLite does not order rows by default (in fact, I don't even think return order is guaranteed? 😬), but if `ORDER BY` is passed in to the statement, I think you can sort by one column, either in ascending or descending order, which I think applies to strings as well. Just doing a quick `sshare` on my machine, it looks like it sorts banks by alphabetical order? So I think I could make that change here. 

I think that would involve changing the `s_sub_banks` and `s_assoc` queries in **data_reader_db.cpp**. I tested this out, and I think I was able to get it working. Here is a super quick demo:

```console
[moussa1@docker-desktop src]$ flux account add-bank root 1
[moussa1@docker-desktop src]$ flux account add-bank --parent-bank=root C 1
[moussa1@docker-desktop src]$ flux account add-bank --parent-bank=root B 1
[moussa1@docker-desktop src]$ flux account add-bank --parent-bank=root A 1
[moussa1@docker-desktop src]$ flux shares
Account                         Username           RawShares            RawUsage
root                                                       1                   0
 A                                                         1                   0
 B                                                         1                   0
 C                                                         1                   0
[moussa1@docker-desktop src]$ flux account add-user --username=userA --bank=C
[moussa1@docker-desktop src]$ flux account add-user --username=userZ --bank=C
[moussa1@docker-desktop src]$ flux account add-user --username=userF --bank=C
[moussa1@docker-desktop src]$ flux shares
Account                         Username           RawShares            RawUsage
root                                                       1                   0
 A                                                         1                   0
 B                                                         1                   0
 C                                                         1                   0
  C                                userA                   1                   0
  C                                userF                   1                   0
  C                                userZ                   1                   0
```
So, above, I inserted three sub banks under `root` in reverse alphabetical order: **C**, **B**, **A**. As for the associations, the insert order was **userA**, **userZ**, **userF**. 

I can push up this change in an additional commit in this PR.

Reply:
Awesome! Looks good.

Reply:
Can you add this out-of-order insertion and then in-order printing as a test case (if it isn't already covered by another test)?

Reply:
Sure. I went ahead and added this as another test database that **t1000-print-hierarchy.t** compares output with and pushed it up.